src/main/java/io/ticktok/Listener/TickListener.java

```diff
@@ -33,9 +37,13 @@ public void handleDelivery(String consumerTag, Envelope envelope,
         return connection.createChannel();
     }
 
-    private static Connection createConnection(Channel channel) throws Exception {
-        ConnectionFactory factory = new ConnectionFactory();
-        factory.setUri(new URI(channel.getUri()));
-        return factory.newConnection();
+    private static Connection createConnection(Channel channel) {
+        try {
+            ConnectionFactory factory = new ConnectionFactory();
+            factory.setUri(channel.getUri());
+            return factory.newConnection();
+        } catch (Exception e) {
```

Comment:
why we're catching all exceptions?

Reply:
that's a pain.
it because `ConnectionFactory.setUri` throws all the following exceptions:
`NoSuchAlgorithmException`, `KeyManagementException`, `URISyntaxException`, `IOException`, `InterruptedException`.

and `Exception` is the only base...
have other ideas?

Reply:
yea find a higher common ground or catch all of these explicitly

Reply:
im not sure i understand... can you give me an example?

Reply:
After trying it out it seems that you only need to catch IOEXception,